crypto/evp/evp_lib.c

```diff
@@ -58,8 +66,45 @@ int EVP_CIPHER_param_to_asn1(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
         default:
             ret = EVP_CIPHER_set_asn1_iv(c, type);
         }
-    } else
-        ret = -1;
+    } else {
+        if (cipher->prov != NULL) {
```

Comment:
This could be made into a single line:
````
} else if (cipher->prov != NULL) {
````
This reduces the level of indent for the code below.

Reply:
Good point